frappe/email/doctype/email_account/email_account.py

```diff
@@ -489,31 +491,35 @@ def remove_unpicklable_values(self, state):
 		state.pop("_smtp_server_instance", None)
 
 	def handle_incoming_connect_error(self, description):
-		if self.get_failed_attempts_count() > 2:
-			self.db_set("enable_incoming", 0)
-
-			for user in get_system_managers(only_name=True):
-				try:
-					assign_to.add(
-						{
-							"assign_to": user,
-							"doctype": self.doctype,
-							"name": self.name,
-							"description": description,
-							"priority": "High",
-							"notify": 1,
-						}
-					)
-				except assign_to.DuplicateToDoError:
-					frappe.clear_last_message()
+		if self.get_failed_attempts_count() > 5:
+			# This is done in background to avoid committing here.
+			frappe.enqueue(self._disable_broken_incoming_account, description=description)
 		else:
 			self.set_failed_attempts_count(self.get_failed_attempts_count() + 1)
 
+	def _disable_broken_incoming_account(self, description):
+		self.db_set("enable_incoming", 0)
```

Comment:
@ankush can we set `update_modified=True` here? otherwise it's hard for the user to notice what has changed and when.